tests/performance-test/performance-test.sh

```diff
@@ -49,91 +49,170 @@ ellipse(){
+        in reduced functionality of performance test. Would you like to continue without \
+        it? [y/n]"
+        read RESP
+        case $RESP in
+            "y") 
+            ;;
+            *)
+                exit 1
+            ;;
+        esac
+    fi
+
     if ! GRAF_HOST=$(oc get routes --field-selector metadata.name=grafana -o jsonpath="{.items[0].spec.host}") 2> /dev/null; then 
         echo "Error: cannot find Grafana instance in cluster. Has it been deployed?" 1>&2
         exit 1
     fi
 
-    SAF_PROM_HOST=$(oc get routes --field-selector metadata.name=prometheus -o jsonpath="{.items[0].spec.host}")
+    local datasources=$(curl --silent -X GET "http://${GRAF_HOST}/api/datasources")
+    if ! echo $datasources | grep -q "OCPPrometheus"; then
+        echo "Error: unable to find Grafana datasource OCPPrometheus"
+        exit 1
+    fi
 
-    SAF_DATASOURCE=$(cat << EOF
-        {
-            "name"  :   "SAFPrometheus",
-            "type"  :   "prometheus",
-            "url"   :   "http://$SAF_PROM_HOST",
-            "access":   "direct",
-            "basicAuth" :false,
-            "jsonData": {
-                "timeInterval"  :   "1s"
-            }
-        }
-EOF
-)
+    if ! echo $datasources | grep -q "SAFPrometheus"; then
+        echo "Error: unable to find Grafana datasource SAFPrometheus"
+        exit 1
+    fi
+
+    if ! echo $datasources | grep -q "SAFElasticsearch"; then
+        echo "Error: unable to find Grafana datasource SAFElasticsearch"
+        exit 1
+    fi
 }
 
+# Post dashboards to grafana - overwrites if they already exist
 post_dashboards(){
-    echo "Creating new dashboards in Grafana"
+    printf "\n*** Creating new dashboards in Grafana ***\n"
     curl --silent --output /dev/null -d "{\"overwrite\": true, \"dashboard\": $(cat ./grafana/perftest-dashboard.json)}" \
         -H 'Content-Type: application/json' "$GRAF_HOST/api/dashboards/db"
         
     curl --silent --output /dev/null -d "{\"overwrite\": true, \"dashboard\": $(cat ./grafana/prom2-dashboard.json)}" \
         -H 'Content-Type: application/json' "$GRAF_HOST/api/dashboards/db"
 }
 
-# Execute
+# Delete recorded events in Elastic Search
+delete_es_events(){
+    local esPod=$(oc get elasticsearch elasticsearch -o jsonpath='{.status.pods.master.ready[0]}')
+    local secretPath="/etc/elasticsearch/secret/"
+    local esQuery=$(cat <<EOF
+        {
+            "query" : {
+                "bool": {
+                    "must": {
+                        "match_phrase": {
+                            "labels.instance": {
+                                "query": "saf-perftest-notify-*"
+                            }
+                        }
+                    }
+                }
+            }
+        }
+EOF
+)
+    printf "\n*** Deleting ES Events ***\n"
+
+    local response=$(oc exec "$esPod" -- curl -k --silent --trace --cacert "${secretPath}admin-ca" --cert \
```

Comment:
`oc exec "$esPod" -- curl -k --silent --trace....` appears 4 times so deserves to be it's own function. Not worth blocking on, but something to note for a potential future refactor.

Reply:
How would the function account for different arguments and queries? For example some of them I have to POST while others are a DELETE